### PR TITLE
ROC-6114 Initial commit

### PR DESCRIPTION
**Pull request by andrewwa-kainos**
### JIRA link ###
https://tools.hmcts.net/jira/browse/ROC-6114

### Change description ###
Initial implementation

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/payments/client/models/PaymentDto.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    private String giroSlipNo;
+    private String id;
+    private String method;
+    private String organisationName;
+    private String paymentGroupReference;
+    private String paymentReference;
+    private String reference;
+    private String reportedDateOffline;
+    // at the time of writing, permitted services are CMC, DIVORCE, PROBATE, FINREM and DIGITAL_BAR
+    private String serviceName;
```
Comment (andrewwa-kainos) [inline src/main/java/uk/gov/hmcts/reform/payments/client/models/PaymentDto.java]: Seems to be an `enum` in the payhub code

File: src/functionalTest/java/uk/gov/hmcts/reform/payments/client/functional/env/CreateUserRequest.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+
+@Data
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
+@ToString
```
Comment (simad) [inline src/functionalTest/java/uk/gov/hmcts/reform/payments/client/functional/env/CreateUserRequest.java]: this is already included in `@Data`

File: src/functionalTest/java/uk/gov/hmcts/reform/payments/client/functional/env/IdamService.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@ToString
```
Comment (simad) [inline src/functionalTest/java/uk/gov/hmcts/reform/payments/client/functional/env/IdamService.java]: same, already included in `@Data`

File: src/functionalTest/java/uk/gov/hmcts/reform/payments/client/functional/env/LoginDetails.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package uk.gov.hmcts.reform.payments.client.functional.env;
+
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+
+@Data
+@RequiredArgsConstructor
```
Comment (simad) [inline src/functionalTest/java/uk/gov/hmcts/reform/payments/client/functional/env/LoginDetails.java]: same, already included in `@Data`

File: src/functionalTest/java/uk/gov/hmcts/reform/payments/client/functional/env/LoginResponse.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+import lombok.Builder;
+import lombok.Data;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+
+@Data
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
+@ToString
```
Comment (simad) [inline src/functionalTest/java/uk/gov/hmcts/reform/payments/client/functional/env/LoginResponse.java]: same, already included in `@Data`

File: src/functionalTest/java/uk/gov/hmcts/reform/payments/client/functional/env/RoleDetail.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+
+@Data
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
+@ToString
```
Comment (simad) [inline src/functionalTest/java/uk/gov/hmcts/reform/payments/client/functional/env/RoleDetail.java]: same, already included in `@Data`

File: src/functionalTest/java/uk/gov/hmcts/reform/payments/client/functional/env/UserDetails.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+
+@Data
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
+@ToString
```
Comment (simad) [inline src/functionalTest/java/uk/gov/hmcts/reform/payments/client/functional/env/UserDetails.java]: same, already included in `@Data`

File: src/main/java/uk/gov/hmcts/reform/payments/client/CardPaymentRequest.java [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+import uk.gov.hmcts.reform.payments.client.models.FeeDto;
+
+import java.math.BigDecimal;
+
+@Data
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
+@ToString
```
Comment (simad) [inline src/main/java/uk/gov/hmcts/reform/payments/client/CardPaymentRequest.java]: same, already included in `@Data`

File: src/main/java/uk/gov/hmcts/reform/payments/client/models/LinkDto.java [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import java.net.URI;
+
+@Data
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
+@ToString
```
Comment (simad) [inline src/main/java/uk/gov/hmcts/reform/payments/client/models/LinkDto.java]: same, already included in `@Data`

File: src/main/java/uk/gov/hmcts/reform/payments/client/models/LinksDto.java [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+
+@Data
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
+@ToString
```
Comment (simad) [inline src/main/java/uk/gov/hmcts/reform/payments/client/models/LinksDto.java]: same, already included in `@Data`

File: src/main/java/uk/gov/hmcts/reform/payments/client/models/PaymentDto.java [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+
+import java.math.BigDecimal;
+import java.time.OffsetDateTime;
+
+@Data
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
+@ToString
```
Comment (simad) [inline src/main/java/uk/gov/hmcts/reform/payments/client/models/PaymentDto.java]: same, already included in `@Data`

File: src/main/java/uk/gov/hmcts/reform/payments/client/models/StatusHistoryDto.java [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+
+import java.time.OffsetDateTime;
+
+@Data
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
+@ToString
```
Comment (simad) [inline src/main/java/uk/gov/hmcts/reform/payments/client/models/StatusHistoryDto.java]: same, already included in `@Data`

File: build.gradle [hunk 13]
```diff
@@ -0,0 +1,221 @@
...
+            }
+        }
+    }
+}
+
+def springBootVersion = '2.1.6.RELEASE'
+
+checkstyle {
+    maxWarnings = 0
+    toolVersion = '8.22'
```
Comment (sabahirfan) [inline build.gradle]: would be nice to update to `8.24`

File: build.gradle [hunk 14]
```diff
@@ -0,0 +1,221 @@
...
+
+dependencies {
+    api group: 'org.springframework.cloud', name: 'spring-cloud-starter-openfeign'
+    api group: 'io.github.openfeign', name: 'feign-jackson', version: feignVersion
+
+    compileOnly group: 'org.projectlombok', name: 'lombok', version: lombokVersion
+    annotationProcessor group: 'org.projectlombok', name: 'lombok', version: lombokVersion
+
+    compile group: 'io.springfox', name: 'springfox-swagger2', version: versions.springfoxSwagger
+    compile group: 'io.springfox', name: 'springfox-swagger-ui', version: versions.springfoxSwagger
```
Comment (sabahirfan) [inline build.gradle]: I don't think we need swagger for this project

File: build.gradle [hunk 15]
```diff
@@ -0,0 +1,218 @@
...
+        version {
+            name = project.version
+        }
+    }
+}
+
+def versions = [
+        logback          : '1.2.3',
+        springBoot       : springBootVersion,
+        springfoxSwagger : '2.9.2',
```
Comment (sabahirfan) [inline build.gradle]: please remove this as well.

**Pull request merged**